packages/backend/src/models/HuggingFaceModelHandler.ts

```diff
@@ -0,0 +1,143 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { ModelHandler } from './ModelHandler';
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
+import { Downloader } from '../utils/downloader';
+import { scanCacheDir, snapshotDownload } from '@huggingface/hub';
+import type { CompletionEvent } from './baseEvent';
+import { getDurationSecondsSince } from '../utils/utils';
+import type { ModelsManager } from '../managers/modelsManager';
+import fs from 'node:fs/promises';
+
+function parseURL(url: string): { repo: string; revision?: string } | undefined {
+  const u = URL.parse(url);
+  if (u) {
+    return { repo: u.pathname.slice(1), revision: u.searchParams.get('revision') ?? 'main' };
+  }
+  return undefined;
+}
+
+class HuggingFaceDownloader extends Downloader {
+  #target: string = '';
+
+  constructor(
+    url: string,
+    private repo: string,
+    private revision: string | undefined,
+    private abortSignal: AbortSignal,
+  ) {
+    super(url, '');
+  }
+
+  override getTarget(): string {
+    return this.#target;
+  }
+
+  async perform(id: string): Promise<void> {
+    const startTime = performance.now();
+
+    try {
+      this.#target = await snapshotDownload({
+        repo: this.repo,
+        revision: this.revision,
+      });
+      const durationSeconds = getDurationSecondsSince(startTime);
+      this._onEvent.fire({
+        id: id,
+        status: 'completed',
+        message: `Duration ${durationSeconds}s.`,
+        duration: durationSeconds,
+      } as CompletionEvent);
+    } catch (err: unknown) {
+      if (!this.abortSignal?.aborted) {
+        this._onEvent.fire({
+          id: id,
+          status: 'error',
+          message: `Something went wrong: ${String(err)}.`,
+        });
+      } else {
+        this._onEvent.fire({
+          id: id,
+          status: 'canceled',
+          message: `Request cancelled: ${String(err)}.`,
+        });
+      }
+      throw err;
+    } finally {
+      this.completed = true;
+    }
+  }
+}
+
+export class HuggingFaceModelHandler extends ModelHandler {
+  constructor(modelsManager: ModelsManager) {
+    super('huggingface model registry', modelsManager);
+  }
+
+  accept(url: string): boolean {
+    return url.startsWith('huggingface') || url.startsWith('hf');
+  }
+
+  createDownloader(model: ModelInfo, abortSignal: AbortSignal): Downloader {
+    const result = parseURL(model.url!);
+    if (result) {
+      return new HuggingFaceDownloader(model.url!, result.repo, result.revision, abortSignal);
+    }
+    throw new Error(`Invalid URL: ${model.url} for model ${model.name}`);
+  }
+
+  async deleteModel(model: ModelInfo): Promise<void> {
+    if (model.file) {
+      await fs.rm(model.file?.path, { recursive: true });
+    }
+  }
+
+  dispose(): void {}
+
+  async getLocalModelsFromDisk(): Promise<void> {
```

Comment:
getLocalModelsFromDisk return nothing, so probably should be renamed to reflect its purpose (scan, collect etc.) or return the models

Reply:
Not related to this PR as it's from the ModelHandler interface. Follow up issue